pkg/services/check/check_test.go

```diff
@@ -0,0 +1,168 @@
+package check
+
+import (
+	"context"
+	"errors"
+
+	"github.com/weaveworks/weave-gitops/pkg/flux"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/weaveworks/weave-gitops/pkg/kube/kubefakes"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/weaveworks/weave-gitops/pkg/flux/fluxfakes"
+)
+
+var _ = Describe("Check", func() {
+
+	var fakeFluxClient *fluxfakes.FakeFlux
+	var fakeKubeClient *kubefakes.FakeKube
+	someError := errors.New("some error")
+	var context context.Context
+	const successfulFluxPreCheckOutput = `► checking prerequisites
+✔ Kubernetes 1.21.1 >=1.19.0-0
+✔ prerequisites checks passed`
+	const expectedFluxVersion = "1.21.1"
+	BeforeEach(func() {
+		fakeFluxClient = &fluxfakes.FakeFlux{}
+		fakeKubeClient = &kubefakes.FakeKube{}
+	})
+
+	It("should fail running flux.PreCheck", func() {
+		fakeFluxClient.PreCheckReturns("", someError)
+
+		out, err := Pre(context, fakeKubeClient, fakeFluxClient, expectedFluxVersion)
+		Expect(err.Error()).To(ContainSubstring(someError.Error()))
+		Expect(out).Should(BeEmpty())
+	})
+
+	It("should show flux is not installed", func() {
+
+		fakeFluxClient.PreCheckReturns(successfulFluxPreCheckOutput, nil)
+
+		fakeKubeClient.GetNamespacesReturns(&corev1.NamespaceList{
+			Items: []corev1.Namespace{},
+		}, nil)
+
+		output, err := Pre(context, fakeKubeClient, fakeFluxClient, expectedFluxVersion)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		Expect(output).To(Equal(`✔ Kubernetes 1.21.1 >=1.19.0-0
+✔ Flux is not installed`))
```

Comment:
```suggestion
		Expect(Pre(context, fakeKubeClient, fakeFluxClient, expectedFluxVersion)).To(
			Equal(`✔ Kubernetes 1.21.1 >=1.19.0-0
✔ Flux is not installed`))
```
NIt- could be simplified to a single Expect() call

Reply:
My personal preference is to have the function call and the Expect in the same level of indentation. Do this approach also checks that error is not happening? I couldn't find a docs explaining. For this particular case actually it would be cleaner to move the expected output to the first level of indentation as well so it's more readable.